python/hsfs/feature_view.py

```diff
@@ -418,46 +415,583 @@ def create_training_dataset(
+                statistics computation pass `statistics_config=False`. Defaults to
+                `None` and will compute only descriptive statistics.
+            read_options: Additional read options as key-value pairs, defaults to `{}`.
+                When using the `python` engine, read_options can contain the
+                following entries:
+                * key `spark` and value an object of type
+                [hsfs.core.job_configuration.JobConfiguration](../job_configuration)
+                  to configure the Hopsworks Job used to compute the training dataset.
+
+        # Returns
+            (X_train, y_train, X_val, y_val, X_test, y_test):
+                Tuple of dataframe of features and labels
+
+        """
+
+        self._validate_train_validation_test_splits(
+            val_size=val_size,
+            test_size=test_size,
+            train_end=train_end,
+            val_start=val_start,
+            val_end=val_end,
+            test_start=test_start,
+        )
+        td = training_dataset.TrainingDataset(
+            name=self.name,
+            version=None,
+            splits={},
+            val_size=val_size,
+            test_size=test_size,
+            train_start=train_start,
+            train_end=train_end,
+            val_start=val_start,
+            val_end=val_end,
+            test_start=test_start,
+            test_end=test_end,
+            description=description,
+            storage_connector=None,
+            featurestore_id=self._featurestore_id,
+            data_format="tsv",
+            location="",
+            statistics_config=statistics_config,
+            training_dataset_type=training_dataset.TrainingDataset.IN_MEMORY,
+        )
+        td, df = self._feature_view_engine.get_training_data(
+            self,
+            read_options,
+            training_dataset_obj=td,
+            splits=[
+                TrainingDatasetSplit.TRAIN,
+                TrainingDatasetSplit.VALIDATION,
+                TrainingDatasetSplit.TEST,
+            ],
+        )
+        warnings.warn(
+            "Incremented version to `{}`.".format(td.version),
+            util.VersionWarning,
+        )
+        return (
+            df[TrainingDatasetSplit.TRAIN]
+            + df[TrainingDatasetSplit.VALIDATION]
+            + df[TrainingDatasetSplit.TEST]
+        )
+
+    @staticmethod
+    def _validate_train_validation_test_splits(
+        val_size, test_size, train_end, val_start, val_end, test_start
+    ):
+        if not (
+            (val_size and test_size)
+            or ((train_end or val_start) and (val_end or test_start))
+        ):
+            raise ValueError(
+                "Invalid split input."
+                "You should specify either (`val_size` and `test_size`)"
+                " or ((`train_end` or `val_start`) and (`val_end` or `test_start`))"
+            )
+
+    def get_training_dataset(
```

Comment:
Can you document the behaviour if there is no label defined on the FeatureView?

Reply:
How about set `y` to `None`?